satellite/marketing/marketingweb/server.go

```diff
@@ -128,20 +163,74 @@ func (s *Server) parseTemplates() (err error) {
 	return nil
 }
 
+// Time converter is used by decoder in formToStruct to format expiration date field from form input.
+func timeConverter(value string) reflect.Value {
+	v, err := time.Parse("2006-01-02",value)
+	if err != nil {
+		return reflect.Value{}
+	}
+	return reflect.ValueOf(v)
+}
+
+// formToStruct is used by createOffer to format post form input into a newOffer.
+func formToStruct(w http.ResponseWriter, req *http.Request) (o marketing.NewOffer, e error){
+	err := req.ParseForm()
+	if err != nil {
+		return o, err
+	}
+	defer req.Body.Close()
+
+	if err := decoder.Decode(&o, req.PostForm);err != nil {
+		return o, err
+	}
+	return o,nil
+}
+
+// Handler for POST requests to create, insert, and start a new offer or credit.
```

Comment:
This is still not following the comment pattern // <function name> .....

Reply:
pushed